docs/manual/extend.md

```diff
@@ -16,3 +10,80 @@ find_package(remage REQUIRED)
 # add_executable(myapp ...)
 target_link_libraries(myapp PRIVATE RMG::remage)
 ```
+
+## Forwarding arguments
+
+The {cpp:class}`RMGManager` is crucial for any of your code to extend _remage_.
+You can access it via
+
+```c++
+#include "RMGManager.hh"
+/*...*/
+RMGManager manager("myapp", argc, argv);
+```
+
+Using the {cpp:class}`RMGManager` you can forward the properties that you
+otherwise would have forwarded through CLI arguments:
+
+```c++
+manager.IncludeMacroFile(macroName);
+manager.SetInteractive(true);
+manager.SetOutputFileName(outputfilename);
+manager.SetNumberOfThreads(nthreads);
+manager.SetOutputOverwriteFiles(overwrite);
+```
+
+For all of the possible options check the public functions of the
+{cpp:class}`RMGManager`.
+
+## Registering custom classes
+
+To register your own custom C++ classes within _remage_ you can use some
+pre-defined methods of _remage_. They are distributed in two places: as class
+methods of either {cpp:class}`RMGManager` or {cpp:class}`RMGUserInit`.
+
+```c++
+manager.SetUserInit(new MyGeometry());
+manager.SetUserInit(new MyPhysicsList());
+```
+
+The `SetUserInit()` method supports a detector geometry inheriting from
+{cpp:class}`RMGHardware`, a run manager inheriting from `G4RunManager`, a
+visualization manager inheriting from `G4VisManager` and a custom physics list
+inheriting from `G4VUserPhysicsList`.
+
+If you want to register more custom classes, you will have to use the
+{cpp:class}`RMGUserInit` class. Access it with
+`auto user_init = RMGManager::Instance()->GetUserInit();`. The obtained instance
+now lets you register more custom classes that can customize the simulation flow
+more than macro commands:
+
+```c++
+user_init->AddSteppingAction<T>(/*...*/);
+user_init->AddTrackingAction<T>(/*...*/);
+user_init->SetUserGenerator<T>(/*...*/);
+user_init->AddOutputScheme<T>(/*...*/);
+user_init->AddOptionalOutputScheme<T>("name", /*...*/);
+user_init->ActivateOptionalOutputScheme("name");
+```
+
+Where you replace `T` with the name of your class and `/*...*/` with the
+arguments your class constructor would take when doing `new T(/*...*/)`. Your
+stepping action should inherit from `G4UserSteppingAction`, the tracking action
+from `G4UserTrackingAction`, the generator from {cpp:class}`RMGVGenerator` and
+your custom output scheme from {cpp:class}`RMGVOutputScheme`. For more info
+about {cpp:class}`RMGUserInit` check the API docs.
+
+## Starting the run
+
+After setting up your C++ code and classes, you will have to start _remage_. You
+do that with
+
+```c++
+manager.Initialize();
+manager.Run();
+```
+
+This means after {cpp:func}`RMGManager::Run` you can add more custom C++ code
```

Comment:
ah, I looked into it again. this is slightly weirdly named: The RMGManager::Run does not call G4RunManager::Run

and RMGManager::Initialize() does not call G4RunManager::Initialize()

In the current architecture macro files are _required_ as those are actually calling these via `/run/initialize` and `/run/beamOn`

Reply:
yes.
* `RMGManager::Initialize` sets up physics list _instances_ etc. (-> this registers the messenger commands for them etc)
* `RMGManager::Run` executes macro or interactive session

* `G4RunManager::Initialize()` or `/run/initialize` then actually intiailizes the processes, geometry etc.
* And `/run/beamOn` _actually_ starts the run.